### PR TITLE
RPC Module

### PR DESCRIPTION
**Pull request by ajamaica**
Adds RPC module
Adds Account data objects
Add serializers for Solana Accounts
Add Support for Borsh serializers
Adds Networking support with a default networking implementation
Adds RpcInterface with only AccountInfo Support

### CODE REVIEW

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/Rpc.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     * Example:
+     * ```
+     * val accountInfo = getAccountInfo(
+     *     randomPublicKey,
+     *     null,
+     *     serializer = BorshAsBase64JsonArraySerializer(Metadata.serializer())
+     * )
+     * ```
+     */
+    override suspend fun <T>getAccountInfo(
```
Comment (creativedrewy) [inline rpc/src/commonMain/kotlin/foundation/metaplex/rpc/Rpc.kt:54]: Nit: Space should be between generic and function name

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    val commitment: Commitment? = null,
+    val minContextSlot: UInt? = null,
+    val dataSlice: RpcDataSlice? = null
+)
+
+/**
+ * Enumeration representing the commitment level for RPC requests.
+ */
+@Serializable
+enum class Commitment {
```
Comment (creativedrewy) [inline rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt:42]: I might advocate for putting all of these data classes in a file alongside the interface. Just to keep definitions clear. I can be talked out of this though.

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/networking/NetworkDriver.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    override suspend fun makeHttpRequest(request: HttpRequest): String {
+        val response = httpClient.request(request.url) {
+            // Set HTTP request properties, including headers and method
+            request.properties.forEach { (key, value) ->
+                headers.append(key, value)
+            }
+            method = HttpMethod(request.method)
+            // Workaround for RPC Response
+            // Removes a specific portion from the request body if present. RPCs complain about it
+            setBody(request.body?.replace("\"type\":\"com.funkatronics.rpccore.JsonRpc20Request\",", ""))
```
Comment (creativedrewy) [inline rpc/src/commonMain/kotlin/foundation/metaplex/rpc/networking/NetworkDriver.kt:38]: I don't know all the details of this implementation, but hard-coding this string feels prone to problems in the future, esp if this becomes a solana lib in the future. Any way to not need the string?

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/serializers/AnchorAccountSerializer.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import diglol.crypto.Hash
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.serializer
+
+open class DiscriminatorSerializer<T>(val discriminator: ByteArray, serializer: KSerializer<T>)
```
Comment (creativedrewy) [inline rpc/src/commonMain/kotlin/foundation/metaplex/rpc/serializers/AnchorAccountSerializer.kt:11]: I'm not familiar with the concept of a "Discriminator". Is this a Solana thing? Can you add a description?

**Review by creativedrewy — APPROVED** [review]
Just some small stuff, but otherwise looking good

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/networking/NetworkDriver.kt (hunk 3 above)
Comment (ajamaica) [reply rpc/src/commonMain/kotlin/foundation/metaplex/rpc/networking/NetworkDriver.kt:38]: This is a work arround from a “bug” on the rpccore lib

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/serializers/AnchorAccountSerializer.kt (hunk 4 above)
Comment (ajamaica) [reply rpc/src/commonMain/kotlin/foundation/metaplex/rpc/serializers/AnchorAccountSerializer.kt:11]: I added more docs specially for anchor

File: rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt (hunk 2 above)
Comment (ajamaica) [reply rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt:42]: Arent they in the same file?

Comment (creativedrewy) [reply rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt:42]: Yeah I just mean all those little helper classes should be in a _different_ file from the RpcInterface definition. Just for cleanliness.

**Pull request merged**